feat: Todo application

richajituri1 opened the pull request:
Explain why this change is being made: -
- Flutter application which displays Todo and allows user to view the Todo in ascending as well as descending order

richajituri1 committed:
feat: Todo application

Explain why this change is being made: -
- Flutter application which displays Todo and allows user to view the
  Todo in ascending as well as descending order

richajituri1 committed:
feat: Todo Application

Explain why this change is being made: -

- Application which display Todo and allows user to view Todo in ascending and descending order

Linked JIRA: -

- [QID-672](https://qidp.atlassian.net/browse/QID-672)

RichaJituri committed:
Changes made

richajituri1 commented on lib/clickable_todo_item.dart:
```diff
@@ -0,0 +1,50 @@
+import 'package:flutter/material.dart';
+
+enum Priority { urgent, normal, low }
```
Made the changes and pushed the code

richajituri1 commented on lib/clickable_todo_item.dart:
```diff
@@ -0,0 +1,50 @@
+enum Priority { urgent, normal, low }
+
+class CheckableTodoItem extends StatefulWidget {
+  const CheckableTodoItem(this.text, this.priority, {super.key});
```
Made the changes, used named arguments

richajituri1 commented on lib/todo_item.dart:
```diff
@@ -0,0 +1,34 @@
+enum Priority { urgent, normal, low }
+
+class TodoItem extends StatelessWidget {
+  const TodoItem(this.text, this.priority, {super.key});
```
Used named arguments

KaustubhDoshaheen commented on lib/keys.dart:
```diff
@@ -0,0 +1,81 @@
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
```
@richajituri1 @RichaJituri  Please do not mark comments resolved but Add Comment saying `Resolved`

richajituri1 replied:
resolved

KaustubhDoshaheen commented on lib/keys.dart:
```diff
@@ -0,0 +1,71 @@
+}
+
+class _KeysState extends State<Keys> {
+  var _order = 'asc';
```
shift this isolated string to constants

richajituri1 replied:
resolved

KaustubhDoshaheen commented on lib/keys.dart:
```diff
@@ -0,0 +1,71 @@
+    final sortedTodos = List.of(Constants.todos);
+    sortedTodos.sort((a, b) {
+      final bComesAfterA = a.text.compareTo(b.text);
+      return _order == 'asc' ? bComesAfterA : -bComesAfterA;
```
same

richajituri1 replied:
resolved

KaustubhDoshaheen commented on lib/keys.dart:
```diff
@@ -0,0 +1,71 @@
+
+  void _changeOrder() {
+    setState(() {
+      _order = _order == 'asc' ? 'desc' : 'asc';
```
same

richajituri1 replied:
resolved

KaustubhDoshaheen commented on lib/keys.dart:
```diff
@@ -0,0 +1,71 @@
+      icon: Icon(
+        _order == 'asc' ? Icons.arrow_downward : Icons.arrow_upward,
+        ),
+        label: Text('Sort ${_order == 'asc' ? 'Descendin' : 'Ascending'}'),
```
same sift strings to constants and access that over here

richajituri1 replied:
resolved

KaustubhDoshaheen commented on lib/keys.dart:
```diff
@@ -0,0 +1,71 @@
+    return TextButton.icon(
+      onPressed: _changeOrder, 
+      icon: Icon(
+        _order == 'asc' ? Icons.arrow_downward : Icons.arrow_upward,
```
same

richajituri1 replied:
resolved

KaustubhDoshaheen commented on lib/keys.dart:
```diff
@@ -0,0 +1,71 @@
+        Expanded(
+          child: Column(
+            children: [
+              for (final todo in _orderedTodos)
```
consider creating a function that performs this looping logic and return widget, or consider using list view builder

richajituri1 replied:
resolved

KaustubhDoshaheen commented on lib/main.dart:
```diff
@@ -0,0 +1,26 @@
+      theme: ThemeData(useMaterial3: true),
+      home: Scaffold(
+        appBar: AppBar(
+          title: const Text('Flutter Internals'),
```
same

richajituri1 replied:
resolved

KaustubhDoshaheen commented on lib/todo_item.dart:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    return Padding(
+      padding: const EdgeInsets.all(8),
```
same

richajituri1 replied:
resolved

KaustubhDoshaheen commented on lib/todo_item.dart:
```diff
@@ -0,0 +1,38 @@
+      child: Row(
+        children: [
+          Icon(icon),
+          const SizedBox(width: 12),
```
same

richajituri1 replied:
resolved

richajituri1 committed:
feat: Todo Application

Explain why this change is being made: -

- Todo application which displys Todo and allows user to view in ascending and descending order

#Linked JIRA: -

#- [QID-672](https://qidp.atlassian.net/browse/QID-672)

richajituri1 committed:
Merge branch 'section7' of github.com:richajituri1/flutter-course into section7
